luaui/Widgets/cmd_bar_hotkeys.lua

```diff
@@ -33,35 +25,86 @@ local function reloadWidgetsBindings()
-	Spring.SendCommands("unbindall")
+-- Map old, deleted lua format presets onto uikeys presets
+local legacyToTxt = {
+	['luaui/configs/bar_hotkeys.lua']          = 'luaui/configs/hotkeys/default_keys.txt',
+	['luaui/configs/bar_hotkeys_mnemonic.lua'] = 'luaui/configs/hotkeys/mnemonic_keys.txt',
+	['luaui/configs/bar_hotkeys_60.lua']       = 'luaui/configs/hotkeys/default_keys_60pct.txt',
+	['luaui/configs/bar_hotkeys_grid.lua']     = 'luaui/configs/hotkeys/grid_keys.txt',
+	['luaui/configs/bar_hotkeys_grid_60.lua']  = 'luaui/configs/hotkeys/grid_keys_60pct.txt',
+	['bar_hotkeys_custom.lua']                 = 'uikeys.txt',
+}
 
-	currentLayout = Spring.GetConfigString("KeyboardLayout", 'qwerty')
-	currentKeybindingsFile = Spring.GetConfigString("KeybindingFile", "luaui/configs/bar_hotkeys.lua")
 
-	loadBindings()
+-- TODO: This code is exclusively to convert from the placeholder lua format to the more stable uikeys format
+-- TODO: After a period of ~6 months it should be deleted (around December 2023 or later)
+local function replaceLegacyPreset()
+	local keyFile = Spring.GetConfigString("KeybindingFile")
+	if not keyFile then return false end
 
-	reloadWidgetsBindings()
+	local newFormat = legacyToTxt[keyFile]
+	if not newFormat then return false end
+
+	-- in case the user has a uikeys file already, we need to back it up because it could potentially get overwritten by the following steps
+	if VFS.FileExists("uikeys.txt") then
+		Spring.Echo("BAR Hotkeys: Found existing unused uikeys file, creating a backup called uikeys_auto_backup.txt")
+		Spring.SendCommands("keyreload")
+		Spring.SendCommands("keysave uikeys_auto_backup.txt")
+	end
+
+	-- output the current custom .lua bindings into a uikeys.txt file
+	if keyFile == 'bar_hotkeys_custom.lua' then
+		Spring.Echo("BAR Hotkeys: bar_hotkeys_custom.lua found. This format is deprecated, a " .. newFormat .. " file was written to your bar folder")
+		if VFS.FileExists(keyFile) then
+			Spring.SendCommands("unbindall")
+			loadBindingsLegacy(keyFile)
+		end
+
+		Spring.SendCommands("keysave " .. newFormat)
+	else
+		Spring.SendCommands("keyreload " .. newFormat)
+	end
+
+	Spring.SetConfigString("KeybindingFile", newFormat)
+
+	return true
 end
 
-local function copyDefaultCustom()
-	if VFS.FileExists("bar_hotkeys_custom.lua") then return end
+local function reloadBindings()
+	-- Second parameter here is just a fallback if this config is undefined
+	currentLayout = Spring.GetConfigString("KeyboardLayout", 'qwerty')
+
+	local hasLegacy = replaceLegacyPreset()
+
+	currentKeybindingsFile = Spring.GetConfigString("KeybindingFile", keyConfig.keybindingPresets["Default"])
 
-	local fileToCopy = Spring.GetConfigString("KeybindingFile", "luaui/configs/bar_hotkeys.lua")
-	fileToCopy = (fileToCopy ~= "bar_hotkeys_custom.lua") and fileToCopy or "luaui/configs/bar_hotkeys.lua"
+	if not hasLegacy then
+		if VFS.FileExists(currentKeybindingsFile) then
+			Spring.SendCommands("keyreload " .. currentKeybindingsFile)
```

Comment:
Again, we need to `unbindall` before reloading, because the preset keybinding files don't themselves call `unbindall`